frontend/src/metabase/components/EntityMenuItem/EntityMenuItem.unit.spec.js

```diff
@@ -41,5 +41,28 @@ describe("EntityMenuItem", () => {
         expect(screen.getByTestId("entity-menu-link")).toBeInTheDocument();
       });
     });
+
+    it("should not render if both action and link props are present", () => {
+      render(
+        <EntityMenuItem
+          title="A pencil icon"
+          icon="pencil"
+          link="/derp"
+          action={() => ({})}
+        />,
+      );
+
+      expect(queryIcon("pencil")).not.toBeInTheDocument();
+      expect(screen.queryByText("A pencil icon")).not.toBeInTheDocument();
+      expect(screen.queryByTestId("entity-menu-link")).not.toBeInTheDocument();
+    });
+
+    it("should not render if neither action nor link props are present", () => {
+      render(<EntityMenuItem title="A pencil icon" icon="pencil" />);
+
+      expect(queryIcon("pencil")).not.toBeInTheDocument();
+      expect(screen.queryByText("A pencil icon")).not.toBeInTheDocument();
+      expect(screen.queryByTestId("entity-menu-link")).not.toBeInTheDocument();
+    });
```

Comment:
@ranquild is this an acceptable way of testing that this component didn't even render? I am not aware of any pattern that I could use to make sure that the `render(<EntityMenuItem...>)` equals `null`.

@npfitz suggested that I add tests for the parent `EntityMenu` and to make sure some of these items don't render at all. I might try to do that in a separate PR because it seems like a non-trivial setup.

Will these two tests suffice for now?

Reply:
@nemanjaglumac you can wrap the component in a parent component and use [`toBeEmptyDOMElement` assertion](https://github.com/testing-library/jest-dom#tobeemptydomelement):

```tsx
render(
  <div data-testid="container">
    <EntityMenuItem />
  </div>
)

expect(screen.getByTestId("container")).toBeEmptyDOMElement();
```

Reply:
Awesome!
Thank you @kulyk